service/provider/src/main/java/gov/va/vro/service/provider/bip/service/BipApiService.java

```diff
@@ -52,8 +54,8 @@
  *
  * @author warren @Date 10/31/22
  */
-// TODO: re-enable this when it works
-// @Service
+@Service
+@Conditional(BipConditions.HigherEnvCondition.class)
```

Comment:
The way, I set it up, and we can change it if we want, is that if the env is any of "local", "test", "end2end-test" then the mock API is used. Otherwise, the "real" BIP API is used. The tests are only expected to pass with Mock because otherwise, we cannot control the responses. We can add "dev" in the list if you want, but there is no possibility for the tests to pass in an environment where the "real" BIP API is active.

Reply:
@dimitri-amida is this a response to my comment or @wlin2019 ? This comment is to use  https://www.baeldung.com/spring-profiles directly(@Profile instead of @Conditional). I have not tried this currently but you should be able to use basic logical operators to directly select the profile. So one annotation will be @Profile("local", "test", "end2end-test") and the other will be @Profile("!local & !test & !end2endtest"). That way you do not need to add the code that you did. You might have to play with the syntax here maybe but it will work I think.

We cannot add dev to the list. That beats the purpose. I agree that your test cannot pass in dev.

Reply:
@au2 I am explaining that whether we use @Profile or @Conditional the behavior will be the same. Furthermore, @Profile is weaker than @Conditional because it is actually implemented using @Conditional. I have already tried @Profile, which is a "simpler" solution and it does not work as expected. The implementation I provided works and is more flexible.